### ISSUE TITLE
Message Layout Bug

### ISSUE THREAD

**Issue by braindrain92** [MessageKit/MessageKit#1156]
I copied over code from the MessageStyle section of the example project. It looks like there is a layout issue involving the message cell width for cells without a tail. What is the best way to fix this?

![IMG_3045](https://user-images.githubusercontent.com/49700479/62483720-3cbf5080-b76d-11e9-89e8-0ea07d6853f6.jpg)

**Comment by JulienKode** [comment]
I do not think it's a bug, because the two message have the same size it's just a different design. They also have the same padding with the avatar. Making the `Test`  message longer will make avatars unaligned

**Comment by BrandonZacharie** [comment]
I also agree that it's not a bug but how can the layout be configured so that the first bubble's right side is aligned with the second bubble's right side before the tail?

**Comment by nathantannar4** [comment]
You would need to add extra message bubble inset. Set using the layout of the `MessagesCollectionView`. See example. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by mknippen** [comment]
As someone new to using the project, while it's probably not a bug, it is a very strange default to have. iMessage, Messenger, and many others have them lined up, and it would look more visually appealing if they were lined up. 

Would it be possible to make it lined up the default size, and still allow someone to move via customization?

Two possible solutions would be:
1. Have the bubble tail extend further out, so the end of the bubbles match. If the user wants to move them over more, they can do it with the delegate.
2. Have the standard bubble extend out a few more pixels. 

Either of these would just be modifying the default options, so it would look better to someone dropping in the SDK.

**Comment by nathantannar4** [comment]
I understand where you are coming from but this is not a bug. As a framework MessageKit tries to make less assumptions that could lead to additional complications. To enable this by default we would have to override a messages inset when we detect different message styles.

If you want to mix styles you will also need to be responsible for setting the appropriate inset based on the message style you are assigning. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by mknippen** [comment]
I agree it's not a bug. It is, however, a bad default. For users who want a simple plug and play, this should just work.

I think if we did either of the two options above, we would not have to make any assumptions, or calculations. All bubbles should have a default end spot. The tails should adjust where the bars end. If I add a tail, it should not move my bubble, which is what it currently does.

Instead, the bubble should be put in the same spot (regardless of tail), and the tail should be added to the right of where the bubble is.

In other words, let's move the right justification from the furthest part to the right, and instead have it to the end of the bubble, ignoring the added tail.

**Comment by mknippen** [comment]
Perhaps instead of calling this a bug, could we ask the question: 

"Is this a good default experience?"

**Comment by mknippen** [comment]
If the devs choose not to do this, could we at least post the line of code for users who search with this same problem so they can easily reference how this would be done? This seems like a pretty common use-case.

**Comment by mknippen** [comment]
@nathantannar4 @BrandonZacharie any thoughts here?

**Comment by BrandonZacharie** [comment]
@mknippen I agree with you, fwiw. However, I also agree this is not a bug and since it appears that the direction of this project is to remain less opinionated, I propose some example be added to the README (or wherever appropriate), front and center, that demonstrates how to implement this design. This design is what I imagine many, if not most, will want right out of the box; it would really help if instructions were right where one goes to learn basic implementation details.

**Comment by mknippen** [comment]
@BrandonZacharie that makes sense. Some clear way to indicate it, or having it the default if that's what most want with an easy way to revert back.

**Comment by monstermac77** [comment]
I believe I have the code to fix this. It does involve creating a custom cell, however. https://github.com/MessageKit/MessageKit/issues/1385#issuecomment-1480355202

![Simulator Screen Shot - iPhone 14 Pro - 2023-03-22 at 18 45 31](https://user-images.githubusercontent.com/3255298/227055535-87e48570-dbd4-4b49-bac1-fa9476fa3b7b.png)
